### DISCUSSION TITLE
Preference between gun name accuracy or convenience?

### DISCUSSION THREAD

**Discussion by printomgwhy** [cataclysmbn/Cataclysm-BN#2907] [Q&A] [answered]
I originally posted this on Reddit but it was suggested I do it here instead:

When it comes to real life objects is keeping their in-game names consistent (and therefore more legible and convenient for players/devs) or is keeping their in-game names accurate to real-life more important?

PR #2896 changes the names of the HK416/417 rifles to the H&K416, H&K417, etc. The PR says this is for consistency and to correct typos.

There's a bit of a problem here: the original names (HK416, HK417) weren't typos. Those were their actual model names. The full name of the 416 is "Heckler & Koch HK416", or if you prefer "H & K HK416". Silly, I know, but I'm not a German marketing guy.

So if the goal is accuracy to real-life objects #2896 doesn't do that, it actually does the exact opposite. But if the goal is consistency for gamey purposes or dev purposes it works quite well: for my own curiosity purposes digging through the code or through debug item lists I've certainly noticed the mild annoyance of having to search for both the 'HK' (for the 416/417 series) and 'H&K' (for the MP5, MP7, the imaginary G80, etc.) strings. So what's preferred?

(Personally I lean toward reverting the names back to their original accurate names but I'm just some rando)

~

On a mild tangent, I'm curious as to whether the HK416 is really necessary, and this is coming from a nerd who normally always wants MOAR GUNZ. It's not commonly issued in the US Army and the Marines instead use the M27 IAR, a heavy 16" barrel version of the HK416 and which is also in the game. The HK416 & M27's in-game stats, are nearly identical. Their volume in particular is identical which leads me to believe the HK416's stats are also based on the 16" barrel model.

**Comment by scarf005** [comment, marked as answer]
The reply is a bit late, but here it goes:

> But if the goal is consistency for gamey purposes or dev purposes it works quite well

I think consistency would be preferred. Take [`.223` <-> `5.56mm`](https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2691) or [`.308` <-> `7.62mm`](https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2724) for example, these are basically the same rounds with only stats differing slightly, however they had a negative impact on UX because you had to search it twice.

> I'm curious as to whether the HK416 is really necessary, ...It's not commonly issued in the US Army and the Marines instead use the M27 IAR, a heavy 16" barrel version of the HK416 and which is also in the game. 

I agree. I'm a bit biased against gun diversity (i play with generic guns), but generally, if two items only differ slightly in stats and has no differences, then it would be better unified as single item.
